src/app/charts/line-graph.component.ts

```diff
@@ -216,34 +216,34 @@ export class LineGraphComponent {
             let x2 = this.xRange[0];
             let y = D3.max(this.yData);
 
-            if (this.min && this.minVal) {
+            if (this.min) {
                 // Draw min threshold line
                 let minData = [{'date': x1, 'value': this.minVal}, {'date': x2, 'value': this.minVal}];
                 this.drawLine(minData, 'min-threshold');
 
                 // Prepare data for bar graph
-                let minBars = _.map(this.extractedData, (datum, index) => {
-                    let val = this.minVal >  this.yData[index] ? y * 2 : 0;
-                    return { 'date': datum['date'], 'value': val };
-                });
+                let minBars = _(this.extractedData)
+                              .map((datum, index) => ({ 'date': datum['date'] }))
+                              .filter((bar, index) => this.minVal > this.yData[index])
+                              .value();
 
                 // Add bar graph
-                this.drawBarGraph(minBars, 'min-bar');
+                this.drawBands(minBars, 'min-bar');
             }
 
-            if (this.max && this.maxVal) {
+            if (this.max) {
```

Comment:
Having `&& this.maxVal` had the effect of hiding the bars when `maxVal` was set to zero, as it is falsey. If I leave `&& this.maxVal` in, the line still shows in inappropriate places when it is negative.

This should really be comparing `maxVal` and `minVal` against the `y` range of the chart. I'll look into adding this.